src/frontend/sharedComponents/PasswordInput.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import { Text, StyleSheet, StyleProp, ViewStyle } from "react-native";
+
+import {
+  CodeField,
+  Cursor,
+  useBlurOnFulfill,
+  useClearByFocusCell,
+  MaskSymbol,
+  isLastFilledCell,
+  RenderCellOptions,
+} from "react-native-confirmation-code-field";
+import { SecurityContext } from "../context/SecurityContext";
+import { MEDIUM_GREY, DARK_GREY } from "../lib/styles";
+
+const CELL_COUNT = 5;
+const onlyNumRegEx = new RegExp("^[0-9]+$");
+
+interface PasswordInputProps {
+  handleError: () => void;
+  handleCorrectOrNewPass: (inputtedValue: string, clear: () => void) => void;
+  clearError: () => void;
+  stylesProps?: StyleProp<ViewStyle>;
+}
+
+export const PasswordInput = ({
```

Comment:
couple of bugs/ux issues i'm noticing:

- the inputs aren't vertically centered, which is strangely distracting 😅 
- thoughts on autofocusing this on initial render? or providing a prop to have that option?
- can't delete the first digit using the delete button
- not a fan of clearing the input and blurring when it's determined that it's incorrect
- feeling mixed on the library's default behavior where pressing a cell clears the value of the cells after it - guess not as big of an issue but kind of annoying if i just wanted to fix a single digit

Reply:
Auto focus, vertically centered, and first digit error fixed. I didn't address the other 2, mainly because it is default behaviour and seems more personal preference, then technical :)